packages/realm_dart/lib/src/app.dart

```diff
@@ -219,16 +219,17 @@ class App implements Finalizable {
 
   /// Returns the current value of the base URL used to communicate with the server.
   @experimental
-  Uri? get baseUrl {
-    return Uri.tryParse(realmCore.getBaseUrl(this) ?? '');
+  Uri get baseUrl {
+    return Uri.parse(realmCore.getBaseUrl(this));
   }
 
   /// Temporarily overrides the [baseUrl] value from [AppConfiguration] with a new [baseUrl] value
-  /// used for communicating with the server.
+  /// used for communicating with the server. If set to `null`, the app will revert to the default
+  /// base url.
   ///
```

Comment:
Similar as a previous comment 👍 
```suggestion
  ///
  /// If this operation fails, the app will continue to use the original base URL. If another [App]
  /// operation is started while this function is in progress, that request will use the original
  /// base URL location information.
  ///
```